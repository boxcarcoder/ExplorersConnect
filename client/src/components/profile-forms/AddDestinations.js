import React, { Fragment, useState } from 'react';
import { Link, withRouter } from 'react-router-dom';

//redux
import { addDestinations } from '../../actions/profile';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

const AddDestinations = ({ addDestinations, history }) => {
  //user inputs must have corresponding states
  const [formData, setFormData] = useState({
    hikingTrails: '',
    campSites: '',
    waterAreas: '',
    slopes: '',
    crags: ''
  });

  const { hikingTrails, campSites, waterAreas, slopes, crags } = formData;

  // e.target.name is the name of the input
  // e.target.value is the user input
  const onChange = e => {
    // console.log('[e.target.name]', [e.target.name]);
    // console.log('e.target.value', e.target.value);

    setFormData({
      ...formData,
      [e.target.name]: e.target.value
    });
  };

  const handleSubmit = e => {
    e.preventDefault();
    addDestinations(formData, history);
  };

  return (
    <Fragment>
      <h1 className='large text-primary'>Add Destinations</h1>
      <p className='lead'>
        <i className='fas fa-map-signs text-primary'></i> Add where you've explored during your favorite trips.
      </p>

      <form className='form' onSubmit={handleSubmit}>
        <div className='line'></div>
        <div className='form-group'>
          <h5>Hiking Trails</h5>
          <input
            type='text'
            placeholder='Trail Name'
            name='hikingTrails'
            value={hikingTrails}
            onChange={e => onChange(e)}
          />
        </div>
        <div className='form-group'>
          <h5>Campsites</h5>
          <input
            type='text'
            placeholder='Camp Site'
            name='campSites'
            value={campSites}
            onChange={e => onChange(e)}
          />
        </div>
        <div className='form-group'>
          <h5>Lakes, Rivers, or Oceans</h5>
          <input
            type='text'
            placeholder='Body of Water'
            name='waterAreas'
            value={waterAreas}
            onChange={e => onChange(e)}
          />
        </div>
        <div className='form-group'>
          <h5>Snow Resorts</h5>
          <input
            type='text'
            placeholder='Slopes'
            name='slopes'
            value={slopes}
            onChange={e => onChange(e)}
          />
        </div>
        <div className='form-group'>
          <h5>Crags</h5>
          <input
            type='text'
            placeholder='Rock faces'
            name='crags'
            value={crags}
            onChange={e => onChange(e)}
          />
        </div>

        <div className='vert-m-1'>
          <input type='submit' className='btn btn-primary my-1' />
          <Link className='btn btn-light my-1' to='/dashboard'>
            Go Back
          </Link>
        </div>
      </form>
    </Fragment>
  );
};

AddDestinations.propTypes = {
  addDestinations: PropTypes.func.isRequired
};

export default connect(null, { addDestinations })(withRouter(AddDestinations));
