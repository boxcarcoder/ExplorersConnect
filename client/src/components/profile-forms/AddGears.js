import React, { Fragment, useState } from 'react';
import { Link, withRouter } from 'react-router-dom';

//redux
import { addGears } from '../../actions/profile';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

const AddGears = ({ addGears, history }) => {
  //user inputs must have corresponding states
  const [formData, setFormData] = useState({
    hikeGear: '',
    campGear: '',
    waterGear: '',
    snowGear: '',
    rockClimbingGear: ''
  });

  const {
    hikeGear,
    campGear,
    waterGear,
    snowGear,
    rockClimbingGear
  } = formData;

  // e.target.name is the name of the input
  // e.target.value is the user input
  const onChange = e => {
    setFormData({
      ...formData,
      [e.target.name]: e.target.value
    });
  };

  const handleSubmit = e => {
    e.preventDefault();
    addGears(formData, history);
  };

  return (
    <Fragment>
      <h1 className='large text-primary'>Add Gear</h1>
      <p className='lead'>
        <i className='fas fa-map-signs text-primary'></i> Add all your essential
        gear for an adventure.
      </p>

      <form className='form' onSubmit={handleSubmit}>
        <div className='line'></div>
        <div className='form-group'>
          <h5>Hiking</h5>
          <input
            type='text'
            placeholder='Hiking Gear'
            name='hikeGear'
            value={hikeGear}
            onChange={e => onChange(e)}
          />
        </div>
        <div className='form-group'>
          <h5>Camping</h5>
          <input
            type='text'
            placeholder='Camping Gear'
            name='campGear'
            value={campGear}
            onChange={e => onChange(e)}
          />
        </div>
        <div className='form-group'>
          <h5>Water Sports</h5>
          <input
            type='text'
            placeholder='Water Gear'
            name='waterGear'
            value={waterGear}
            onChange={e => onChange(e)}
          />
        </div>
        <div className='form-group'>
          <h5>Snow Sports</h5>
          <input
            type='text'
            placeholder='Snow Gear'
            name='snowGear'
            value={snowGear}
            onChange={e => onChange(e)}
          />
        </div>
        <div className='form-group'>
          <h5>Rock Climbing</h5>
          <input
            type='text'
            placeholder='Rock Climbing Gear'
            name='rockClimbingGear'
            value={rockClimbingGear}
            onChange={e => onChange(e)}
          />
        </div>

        <div className='vert-m-1'>
          <input type='submit' className='btn btn-primary my-1' />
          <Link className='btn btn-light my-1' to='/dashboard'>
            Go Back
          </Link>
        </div>
      </form>
    </Fragment>
  );
};

AddGears.propTypes = {
  addGears: PropTypes.func.isRequired
};

export default connect(null, { addGears })(withRouter(AddGears));
