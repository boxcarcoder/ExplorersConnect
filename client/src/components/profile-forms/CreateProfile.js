import React, { useState, Fragment } from 'react';
import { withRouter, Link } from 'react-router-dom'; // to redirect from actions using history.push

//redux
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createProfile } from '../../actions/profile';

const CreateProfile = ({ createProfile, history }) => {
  //user inputs must have corresponding states
  const [formData, setFormData] = useState({
    Hiking: false,
    Camping: false,
    Kayaking: false,
    Rafting: false,
    Skiing: false,
    Snowboarding: false,
    Rockclimbing: false,
    Other: false,
    faveRecreation: '',
    website: '',
    bio: '',
    location: '',
    twitter: '',
    facebook: '',
    youtube: '',
    instagram: ''
  });

  const {
    Hiking,
    Camping,
    Kayaking,
    Rafting,
    Skiing,
    Snowboarding,
    Rockclimbing,
    Other,
    faveRecreation,
    website,
    bio,
    location,
    twitter,
    facebook,
    youtube,
    instagram
  } = formData;

  const [displaySocialInputs, toggleSocialInputs] = useState(false);

  // e.target.name is the name of the input
  // e.target.value is the user input
  const onChange = e => {
    setFormData({
      ...formData,
      [e.target.name]: e.target.value
    });
  };

  // e.target.checked is the user clicking the check box
  const onClick = e => {
    console.log('[e.target.name] before: ', [e.target.name]);
    setFormData({
      ...formData,
      [e.target.name]: !e.target.checked
    });
    console.log('[e.target.name] after: ', [e.target.name]);
  };

  // when the form is submitted, fire the createProfile action
  const handleSubmit = e => {
    e.preventDefault();
    console.log('form data from CREATE PROFILE: ', formData);
    createProfile(formData, history, false);
  };

  return (
    <Fragment>
      <h1 className='large text-primary'>Create Your Profile</h1>
      <p className='lead'>
        <i className='fas fa-user'></i> Let's personalize your profile.
      </p>

      <div className='line'></div>

      <form className='form' onSubmit={handleSubmit}>
        {/* Check box list */}
        <p className='lead'>Outdoor Recreations</p>
        <small className='form-text'>
          What outdoor recreations do you enjoy?
        </small>
        <div className='form-group'>
          <input
            type='checkbox'
            name='Hiking'
            value={Hiking}
            onClick={e => onClick(e)}
          />{' '}
          Hiking
          <br />
          <input
            type='checkbox'
            name='Camping'
            value={Camping}
            onClick={e => onClick(e)}
          />{' '}
          Camping
          <br />
          <input
            type='checkbox'
            name='Kayaking'
            value={Kayaking}
            onClick={e => onClick(e)}
          />{' '}
          Kayaking
          <br />
          <input
            type='checkbox'
            name='Rafting'
            value={Rafting}
            onClick={e => onClick(e)}
          />{' '}
          Rafting
          <br />
          <input
            type='checkbox'
            name='Skiing'
            value={Skiing}
            onClick={e => onClick(e)}
          />{' '}
          Skiing
          <br />
          <input
            type='checkbox'
            name='Snowboarding'
            value={Snowboarding}
            onClick={e => onClick(e)}
          />{' '}
          Snowboarding
          <br />
          <input
            type='checkbox'
            name='Rockclimbing'
            value={Rockclimbing}
            onClick={e => onClick(e)}
          />{' '}
          Rockclimbing
          <br />
          <input type='checkbox' name='Other' value={Other} /> Other
          <br />
        </div>

        {/* Favorite recreation dropdown */}
        <small className='form-text'>Which is your favorite recreation?</small>
        <div className='form-group'>
          <select
            name='faveRecreation'
            value={window.faveRecreation}
            onChange={e => onChange(e)}
          >
            <option value='0'>* Favorite Outdoor Recreation</option>
            <option value='HikingFave'>Hiking</option>
            <option value='CampingFave'>Camping</option>
            <option value='KayakingFave'>Kayaking</option>
            <option value='RaftingFave'>Rafting</option>
            <option value='SkiingFave'>Skiing</option>
            <option value='SnowboardingFave'>Snowboarding</option>
            <option value='RockClimbingFave'>Rock Climbing</option>
            <option value='OtherFave'>Other</option>
          </select>
        </div>

        <div className='line'></div>

        <p className='lead'>Bio</p>
        <div className='form-group'>
          {/* Personal Website */}
          <input
            type='text'
            placeholder='Personal Website'
            name='website'
            value={website}
            onChange={e => onChange(e)}
          />
          <small className='form-text'>
            Have a blog for adventures or gear reviews?
          </small>
        </div>

        <div className='form-group'>
          {/* Bio */}
          <textarea
            placeholder='A short bio of yourself'
            name='bio'
            value={bio}
            onChange={e => onChange(e)}
          ></textarea>
          <small className='form-text'>Tell us a little about yourself</small>
        </div>

        <div className='form-group'>
          {/* Location */}
          <input
            type='text'
            placeholder='* Location'
            name='location'
            value={location}
            onChange={e => onChange(e)}
          />
          <small className='form-text'>
            City & state (eg. Los Angeles, CA)
          </small>
        </div>

        <div className='my-2'>
          <button
            onClick={() => toggleSocialInputs(!displaySocialInputs)}
            type='button'
            className='btn'
          >
            Add Social Network Links
          </button>
          <small>Optional</small>
        </div>

        {displaySocialInputs ? (
          <Fragment>
            <div className='form-group social-input'>
              <i className='fab fa-twitter fa-2x'></i>
              {/* Twitter Link */}
              <input
                type='text'
                placeholder='Twitter URL'
                name='twitter'
                value={twitter}
                onChange={e => onChange(e)}
              />
            </div>
            <div className='form-group social-input'>
              <i className='fab fa-facebook fa-2x'></i>
              {/* Facebook Link */}
              <input
                type='text'
                placeholder='Facebook URL'
                name='facebook'
                value={facebook}
                onChange={e => onChange(e)}
              />
            </div>
            <div className='form-group social-input'>
              <i className='fab fa-youtube fa-2x'></i>
              {/* Youtube Link */}
              <input
                type='text'
                placeholder='YouTube URL'
                name='youtube'
                value={youtube}
                onChange={e => onChange(e)}
              />
            </div>
            <div className='form-group social-input'>
              <i className='fab fa-instagram fa-2x'></i>
              {/* Instagram Link */}
              <input
                type='text'
                placeholder='Instagram URL'
                name='instagram'
                value={instagram}
                onChange={e => onChange(e)}
              />
            </div>
          </Fragment>
        ) : null}

        <small>* = required fields</small>
        <div className='vert-m-1'>
          <input type='submit' className='btn btn-primary my-1' />
          <Link className='btn btn-light my-1' to='/dashboard'>
            Go Back
          </Link>
        </div>
      </form>
    </Fragment>
  );
};

CreateProfile.propTypes = {
  createProfile: PropTypes.func.isRequired
};

export default connect(null, { createProfile })(withRouter(CreateProfile));
