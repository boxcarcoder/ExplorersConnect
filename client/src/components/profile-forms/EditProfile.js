import React, { useState, Fragment, useEffect } from 'react';
import { withRouter, Link } from 'react-router-dom'; // to redirect from actions using history.push

//redux
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createProfile, getCurrentProfile } from '../../actions/profile';

const EditProfile = ({
  profile: { profile, loading },
  createProfile,
  history,
  getCurrentProfile
}) => {
  //create state for form data that requires user input
  const [formData, setFormData] = useState({
    Hiking: false,
    Camping: false,
    Kayaking: false,
    Rafting: false,
    Skiing: false,
    Snowboarding: false,
    Rockclimbing: false,
    Other: false,
    faveRecreation: '',
    website: '',
    bio: '',
    location: '',
    twitter: '',
    facebook: '',
    youtube: '',
    instagram: ''
  });

  const {
    Hiking,
    Camping,
    Kayaking,
    Rafting,
    Skiing,
    Snowboarding,
    Rockclimbing,
    Other,
    faveRecreation,
    website,
    bio,
    location,
    twitter,
    facebook,
    youtube,
    instagram
  } = formData;

  const [displaySocialInputs, toggleSocialInputs] = useState(false);

  // run getCurrentProfile to fetch profile data for the state
  useEffect(() => {
    getCurrentProfile();
    console.log('profile data during first load: ', profile);

    // prefill the form data with the current profile values (from getCurrentProfile)
    setFormData({
      Hiking: loading || !profile.Hiking ? false : profile.Hiking,
      Camping: loading || !profile.Camping ? false : profile.Camping,
      Kayaking: loading || !profile.Kayaking ? false : profile.Kayaking,
      Rafting: loading || !profile.Rafting ? false : profile.Rafting,
      Skiing: loading || !profile.Skiing ? false : profile.Skiing,
      Snowboarding:
        loading || !profile.Snowboarding ? false : profile.Snowboarding,
      Rockclimbing:
        loading || !profile.Rockclimbing ? false : profile.Rockclimbing,
      Other: loading || !profile.Other ? false : profile.Other,
      faveRecreation:
        loading || !profile.faveRecreation ? '' : profile.faveRecreation,
      website: loading || !profile.website ? '' : profile.website,
      bio: loading || !profile.bio ? '' : profile.bio,
      location: loading || !profile.location ? '' : profile.location,
      twitter: loading || !profile.social ? '' : profile.social.twitter,
      facebook: loading || !profile.social ? '' : profile.social.facebook,
      linkedin: loading || !profile.social ? '' : profile.social.linkedin,
      youtube: loading || !profile.social ? '' : profile.social.youtube,
      instagram: loading || !profile.social ? '' : profile.social.instagram
    });
  }, [loading]);

  // e.target.name is the name of the input
  // e.target.value is the user input
  const onChange = e => {
    setFormData({
      ...formData,
      [e.target.name]: e.target.value
    });
  };

  // e.target.checked is the user clicking the check box
  const onClick = e => {
    setFormData({
      ...formData,
      [e.target.name]: e.target.checked
    });
  };

  // when the form is submitted, fire the createProfile action
  const handleSubmit = e => {
    e.preventDefault();
    createProfile(formData, history, true);
  };

  return (
    <Fragment>
      <h1 className='large text-primary'>Create Your Profile</h1>
      <p className='lead'>
        <i className='fas fa-user'></i> Let's personalize your profile.
      </p>

      <div className='line'></div>

      <form className='form' onSubmit={handleSubmit}>
        {/* Check box list */}
        <p className='lead'>Outdoor Recreations</p>
        <small className='form-text'>
          What outdoor recreations do you enjoy?
        </small>
        <div className='form-group'>
          <input
            type='checkbox'
            name='Hiking'
            checked={Hiking}
            onClick={e => onClick(e)}
          />{' '}
          Hiking
          <br />
          <input
            type='checkbox'
            name='Camping'
            checked={Camping}
            onClick={e => onClick(e)}
          />{' '}
          Camping
          <br />
          <input
            type='checkbox'
            name='Kayaking'
            checked={Kayaking}
            onClick={e => onClick(e)}
          />{' '}
          Kayaking
          <br />
          <input
            type='checkbox'
            name='Rafting'
            checked={Rafting}
            onClick={e => onClick(e)}
          />{' '}
          Rafting
          <br />
          <input
            type='checkbox'
            name='Skiing'
            checked={Skiing}
            onClick={e => onClick(e)}
          />{' '}
          Skiing
          <br />
          <input
            type='checkbox'
            name='Snowboarding'
            checked={Snowboarding}
            onClick={e => onClick(e)}
          />{' '}
          Snowboarding
          <br />
          <input
            type='checkbox'
            name='Rockclimbing'
            checked={Rockclimbing}
            onClick={e => onClick(e)}
          />{' '}
          Rockclimbing
          <br />
          <input type='checkbox' name='Other' checked={Other} /> Other
          <br />
        </div>

        {/* Favorite recreation dropdown */}
        <small className='form-text'>Which is your favorite recreation?</small>
        <div className='form-group'>
          <select
            name='faveRecreation'
            value={faveRecreation}
            onChange={e => onChange(e)}
          >
            <option value='0'>* Favorite Outdoor Recreation</option>
            <option value='HikingFave'>Hiking</option>
            <option value='CampingFave'>Camping</option>
            <option value='KayakingFave'>Kayaking</option>
            <option value='RaftingFave'>Rafting</option>
            <option value='SkiingFave'>Skiing</option>
            <option value='SnowboardingFave'>Snowboarding</option>
            <option value='RockClimbingFave'>Rock Climbing</option>
            <option value='OtherFave'>Other</option>
          </select>
        </div>

        <div className='line'></div>

        <p className='lead'>Bio</p>
        <div className='form-group'>
          {/* Personal Website */}
          <input
            type='text'
            placeholder='Personal Website'
            name='website'
            value={website}
            onChange={e => onChange(e)}
          />
          <small className='form-text'>
            Have a blog for adventures or gear reviews?
          </small>
        </div>

        <div className='form-group'>
          {/* Bio */}
          <textarea
            placeholder='A short bio of yourself'
            name='bio'
            value={bio}
            onChange={e => onChange(e)}
          ></textarea>
          <small className='form-text'>Tell us a little about yourself</small>
        </div>

        <div className='form-group'>
          {/* Location */}
          <input
            type='text'
            placeholder='* Location'
            name='location'
            value={location}
            onChange={e => onChange(e)}
          />
          <small className='form-text'>
            City & state (eg. Los Angeles, CA)
          </small>
        </div>

        <div className='my-2'>
          <button
            onClick={() => toggleSocialInputs(!displaySocialInputs)}
            type='button'
            className='btn'
          >
            Add Social Network Links
          </button>
          <small>Optional</small>
        </div>

        {displaySocialInputs ? (
          <Fragment>
            <div className='form-group social-input'>
              <i className='fab fa-twitter fa-2x'></i>
              {/* Twitter Link */}
              <input
                type='text'
                placeholder='Twitter URL'
                name='twitter'
                value={twitter}
                onChange={e => onChange(e)}
              />
            </div>
            <div className='form-group social-input'>
              <i className='fab fa-facebook fa-2x'></i>
              {/* Facebook Link */}
              <input
                type='text'
                placeholder='Facebook URL'
                name='facebook'
                value={facebook}
                onChange={e => onChange(e)}
              />
            </div>
            <div className='form-group social-input'>
              <i className='fab fa-youtube fa-2x'></i>
              {/* Youtube Link */}
              <input
                type='text'
                placeholder='YouTube URL'
                name='youtube'
                value={youtube}
                onChange={e => onChange(e)}
              />
            </div>
            <div className='form-group social-input'>
              <i className='fab fa-instagram fa-2x'></i>
              {/* Instagram Link */}
              <input
                type='text'
                placeholder='Instagram URL'
                name='instagram'
                value={instagram}
                onChange={e => onChange(e)}
              />
            </div>
          </Fragment>
        ) : null}

        <small>* = required fields</small>
        <div className='vert-m-1'>
          <input type='submit' className='btn btn-primary my-1' />
          <Link className='btn btn-light my-1' to='/dashboard'>
            Go Back
          </Link>
        </div>
      </form>
    </Fragment>
  );
};

EditProfile.propTypes = {
  createProfile: PropTypes.func.isRequired,
  getCurrentProfile: PropTypes.func.isRequired,
  profile: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  profile: state.profile
});

export default connect(mapStateToProps, { createProfile, getCurrentProfile })(
  withRouter(EditProfile)
);
