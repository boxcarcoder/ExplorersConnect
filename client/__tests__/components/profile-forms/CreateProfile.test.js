import React from 'react';
import { shallow } from 'enzyme';

// Import the non-connected component
import { CreateProfile } from '../../../src/components/profile-forms/CreateProfile';

// Globals for testing
let wrapper;
let props;
let mockCreateProfile = jest.fn();

describe('<CreateProfile /> component.', () => {
    beforeEach(() => {
        props = {
            createProfile: mockCreateProfile,
            history: {
                push: jest.fn()
            }
        };
        wrapper = shallow(<CreateProfile {...props}/>);
    })
    describe('Successfully renders', () => {
        test('the component.', () => {
            expect(wrapper).toMatchSnapshot();
        });

        test('a link to the dashboard.', () => {
            const link = wrapper.find('Link');
            expect(link.props().to).toBe('/dashboard');
        });  

        test('the textboxes for social medias when toggled on.', () => {
            wrapper.find('button').simulate('click');
            // twitter textbox
            expect(wrapper.find('input').at(9).props().name).toBe('twitter');

            // facebook textbox
            expect(wrapper.find('input').at(10).props().name).toBe('facebook');

            // youtube textbox
            expect(wrapper.find('input').at(11).props().name).toBe('youtube');

            // instagram textbox
            expect(wrapper.find('input').at(12).props().name).toBe('instagram');

        })
    });

    describe('The checkbox updates for', () => {
        test('the hiking checkbox.', () => {
            wrapper.find('input').at(0).simulate('change', {
                target: {
                    name: 'Hiking',
                    checked: true,
                }
            })

            // Execute the test
            expect(wrapper.find('input').at(0).props().checked).toBe(true);
        });

        test('the camping checkbox.', () => {
            wrapper.find('input').at(1).simulate('change', {
                target: {
                    name: 'Camping',
                    checked: true,
                }
            })

            // Execute the test
            expect(wrapper.find('input').at(1).props().checked).toBe(true);
        });

        test('the kayaking checkbox.', () => {
            wrapper.find('input').at(2).simulate('change', {
                target: {
                    name: 'Kayaking',
                    checked: true,
                }
            })

            // Execute the test
            expect(wrapper.find('input').at(2).props().checked).toBe(true);
        });

        test('the rafting checkbox.', () => {
            wrapper.find('input').at(3).simulate('change', {
                target: {
                    name: 'Rafting',
                    checked: true,
                }
            })

            // Execute the test
            expect(wrapper.find('input').at(3).props().checked).toBe(true);
        });

        test('the skiing checkbox.', () => {
            wrapper.find('input').at(4).simulate('change', {
                target: {
                    name: 'Skiing',
                    checked: true,
                }
            })

            // Execute the test
            expect(wrapper.find('input').at(4).props().checked).toBe(true);
        });

        test('the snowboarding checkbox.', () => {
            wrapper.find('input').at(5).simulate('change', {
                target: {
                    name: 'Snowboarding',
                    checked: true,
                }
            })

            // Execute the test
            expect(wrapper.find('input').at(5).props().checked).toBe(true);
        });

        test('the rock climbing checkbox.', () => {
            wrapper.find('input').at(6).simulate('change', {
                target: {
                    name: 'Rockclimbing',
                    checked: true,
                }
            })

            // Execute the test
            expect(wrapper.find('input').at(6).props().checked).toBe(true);
        });
    });

    describe('The dropdown updates for', () => {
        test('the favorite outdoor recreation dropdown list.', () => {
            wrapper.find('select').simulate('change', { 
                target: {
                    name: 'faveRecreation',
                    value: 'CampingFave'
                }
            });

            //Execute the test.
            expect(wrapper.find('select').props().value).toBe('CampingFave');
        });
    });

    describe('The textbox updates for', () => {
        test('the personal website textbox.', () => {
            wrapper.find('input').at(7).simulate('change', {
                target: {
                    name:'website',
                    value: 'test website'
                }
            });

            // Execute the test
            expect(wrapper.find('input').at(7).props().value).toBe('test website');
        });

        test('the bio textbox.', () => {
            wrapper.find('textarea').simulate('change', {
                target: {
                    name:'bio',
                    value: 'test bio'
                }
            });

            // Execute the test
            expect(wrapper.find('textarea').props().value).toBe('test bio');
        });

        test('the location textbox.', () => {
            wrapper.find('input').at(8).simulate('change', {
                target: {
                    name:'location',
                    value: 'test location'
                }
            });

            // Execute the test
            expect(wrapper.find('input').at(8).props().value).toBe('test location');
        });
    });

    describe('The textbox updates for the following social medias:', () => {
        beforeEach(() => {
            // Click the display social media button to render them.
            wrapper.find('button').simulate('click');
        });

        test('twitter.', () => {
            wrapper.find('input').at(9).simulate('change', {
                target: {
                    name: 'twitter',
                    value: 'test twitter link'
                }
            });

            // Execute the test.
            expect(wrapper.find('input').at(9).props().value).toBe('test twitter link');
        });

        test('facebook.', () => {
            wrapper.find('input').at(10).simulate('change', {
                target: {
                    name: 'facebook',
                    value: 'test facebook link'
                }
            });

            // Execute the test.
            expect(wrapper.find('input').at(10).props().value).toBe('test facebook link');
        });

        test('youtube.', () => {
            wrapper.find('input').at(11).simulate('change', {
                target: {
                    name: 'youtube',
                    value: 'test youtube link'
                }
            });

            // Execute the test.
            expect(wrapper.find('input').at(11).props().value).toBe('test youtube link');
        });

        test('instagram.', () => {
            wrapper.find('input').at(12).simulate('change', {
                target: {
                    name: 'instagram',
                    value: 'test instagram link'
                }
            });

            // Execute the test.
            expect(wrapper.find('input').at(12).props().value).toBe('test instagram link');
        });
    });

    describe('When the form is submitted', () => {
        test('the default event is prevented.', () => {
            let eventPrevented = false;
            wrapper.find('form').simulate('submit', {
                preventDefault: () => {
                    eventPrevented = true;
                },
            });
        
            expect(eventPrevented).toBe(true);
        });

        test('the createProfile() action is fired.', () => {
            wrapper.find('form').simulate('submit', {
                preventDefault: () => {},
            });        

            expect(mockCreateProfile).toHaveBeenCalled();
        });
    })
});
