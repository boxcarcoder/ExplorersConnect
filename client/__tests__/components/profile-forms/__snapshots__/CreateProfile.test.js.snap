// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateProfile /> component. Successfully renders the component. 1`] = `
<Fragment>
  <h1
    className="large text-primary"
  >
    Create Your Profile
  </h1>
  <p
    className="lead"
  >
    <i
      className="fas fa-user"
    />
     Let's personalize your profile.
  </p>
  <div
    className="line"
  />
  <form
    className="form"
    onSubmit={[Function]}
  >
    <p
      className="lead"
    >
      Outdoor Recreations
    </p>
    <small
      className="form-text"
    >
      What outdoor recreations do you enjoy?
    </small>
    <div
      className="form-group"
    >
      <input
        checked={false}
        name="Hiking"
        onChange={[Function]}
        type="checkbox"
      />
       
      Hiking
      <br />
      <input
        checked={false}
        name="Camping"
        onChange={[Function]}
        type="checkbox"
      />
       
      Camping
      <br />
      <input
        checked={false}
        name="Kayaking"
        onChange={[Function]}
        type="checkbox"
      />
       
      Kayaking
      <br />
      <input
        checked={false}
        name="Rafting"
        onChange={[Function]}
        type="checkbox"
      />
       
      Rafting
      <br />
      <input
        checked={false}
        name="Skiing"
        onChange={[Function]}
        type="checkbox"
      />
       
      Skiing
      <br />
      <input
        checked={false}
        name="Snowboarding"
        onChange={[Function]}
        type="checkbox"
      />
       
      Snowboarding
      <br />
      <input
        checked={false}
        name="Rockclimbing"
        onChange={[Function]}
        type="checkbox"
      />
       
      Rock Climbing
      <br />
    </div>
    <small
      className="form-text"
    >
      Which is your favorite recreation?
    </small>
    <div
      className="form-group"
    >
      <select
        name="faveRecreation"
        onChange={[Function]}
        value=""
      >
        <option
          value="0"
        >
          * Favorite Outdoor Recreation
        </option>
        <option
          value="HikingFave"
        >
          Hiking
        </option>
        <option
          value="CampingFave"
        >
          Camping
        </option>
        <option
          value="KayakingFave"
        >
          Kayaking
        </option>
        <option
          value="RaftingFave"
        >
          Rafting
        </option>
        <option
          value="SkiingFave"
        >
          Skiing
        </option>
        <option
          value="SnowboardingFave"
        >
          Snowboarding
        </option>
        <option
          value="RockClimbingFave"
        >
          Rock Climbing
        </option>
      </select>
    </div>
    <div
      className="line"
    />
    <p
      className="lead"
    >
      Bio
    </p>
    <div
      className="form-group"
    >
      <input
        name="website"
        onChange={[Function]}
        placeholder="Personal Website"
        type="text"
        value=""
      />
      <small
        className="form-text"
      >
        Have a blog for adventures or gear reviews?
      </small>
    </div>
    <div
      className="form-group"
    >
      <textarea
        name="bio"
        onChange={[Function]}
        placeholder="A short bio of yourself"
        value=""
      />
      <small
        className="form-text"
      >
        Tell us a little about yourself
      </small>
    </div>
    <div
      className="form-group"
    >
      <input
        name="location"
        onChange={[Function]}
        placeholder="* Location"
        type="text"
        value=""
      />
      <small
        className="form-text"
      >
        City & state (eg. Los Angeles, CA)
      </small>
    </div>
    <div
      className="my-2"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        Add Social Network Links
      </button>
      <small>
        Optional
      </small>
    </div>
    <small>
      * = required fields
    </small>
    <div
      className="vert-m-1"
    >
      <input
        className="btn btn-primary my-1"
        type="submit"
      />
      <Link
        className="btn btn-light my-1"
        to="/dashboard"
      >
        Go Back
      </Link>
    </div>
  </form>
</Fragment>
`;
